Fixes #13010 by updating boto3 methods with profile_name

Etherdaemon opened the pull request:
Fixes #13010 by updating boto3 methods to have a separate profile_name argument as specified in the proposed fix section of the issue

bcoca committed:
make callbacks declare needing whitelisting
this will allow for custom or v1 callbacks locally loaded to automatically be triggered as they were in 1.x

bcoca committed:
added callback fallback to v1 methods for backwards compat with existing callbacks

amenonsen committed:
Fix typo

Etherdaemon committed:
Fixes #13010 by updating boto3 methods to update boto3_conn to check for
profile_name and if detected, remove from params and set a new variable
of profile which is passed into Session separately

Etherdaemon committed:
Fixes #13010 by updating boto3 methods to have a separate profile_name argument

willthames commented on lib/ansible/module_utils/ec2.py:
```diff
@@ -150,10 +150,8 @@ def get_aws_connection_info(module, boto3=False):
-            boto_params['profile_name'] = profile_name
-
-
+        return region, ec2_url, profile_name, boto_params
```
This appears to change the specification of get_aws_connection_info to return 4 values only when using boto3. 

Wouldn't you need to update every single module that calls `get_aws_connection_info`?

willthames commented on lib/ansible/module_utils/ec2.py:
```diff
@@ -168,8 +166,8 @@ def get_aws_connection_info(module, boto3=False):
             boto_params['validate_certs'] = validate_certs
-
-    return region, ec2_url, boto_params
+        
```
Empty lines should be empty

willthames commented:
Sorry, I commented without reading #13010. I'll comment there instead.

Etherdaemon committed:
Update to only return the extra profile_name if a profile_name was provided in the first place for method get_aws_connection_info

Etherdaemon committed:
Remove empty spaces in line

Etherdaemon committed:
Remove empty spaces in line

Etherdaemon committed:
Update to put get_aws_connection_info back to its original spec, update boto3_conn to check for profile_name and if detected, remove from params and set a new variable of profile which is passed into Session separately

dvarrazzo committed:
Added tests to verify pip installation with VCS urls

The test reproduces:
- bug ansible/ansible#1705
- bug ansible/ansible-modules-core#1645

abadger committed:
Cleanup timer callback plugin's overwriting of modules it imported

abadger committed:
Remove some more unused stdlib deps.

bcoca committed:
fixed bsd service_mgr_facts using removed check_init fucntion

erjohnso committed:
Allow PATCH to list of valid HTTP methods

abadger committed:
remove assigned but unused variables in module_utils.

These were mostly saving exceptions but not using them.  Getting rid of
those will help with eventually running modules via either python2.4 or
python3.x.

smiller171 committed:
Update README.md

abadger committed:
Update submodule refs

Update to a one-liner to assign profile value

Etherdaemon committed:
Merge remote-tracking branch 'upstream/devel' into fix_boto3_aws_profile_13010

Etherdaemon committed:
Update to a one-liner to assign profile value

abadger committed:
Fix up rst formatting

willthames committed:
Allow blank lines and comments/spaces in galaxy spec files

Fixes #10641 again. (Regression)
Added test to discourage future such regression.

bcoca committed:
added note about implicit vs explicit handling of ssh login user

Commit:
treat forks config parameter as max value instead of always creating that number of workers

bcoca committed:
better limiting of forks to lowest actual restriction

bcoca committed:
removed serial as it is not avialable for fork control
fixes #13037

bcoca committed:
use minimal number of forks taking serial, forks and play patterns into account

Commit:
Provide git based example

I found it difficult to find documentation for how to pull from a git-based scm that wasn't github. The only way I could find this option was to dig through the google-group.

bcoca committed:
clearer selection of min value
cleanup workers at each run to avoid stragglers

abadger committed:
Do not optimize with_items loop if the items are not strings

The code isn't sophisticated enough to understand lists and dicts yet.
This mirrors how 1.9.x handled non-string items so its not a regression.

One portion of a fix for #12976

abadger committed:
template the task_action before checking whether to optimize with_item loops for it.

Fixes the second problem discovered in #12976

abadger committed:
Update core submodule ref

jimi-c commented:
@Etherdaemon could you please squash these down to a single commit, and clean up the merges that are in this branch? Thanks!

abadger committed:
Set task_action when the action does not need to be templated

Fixes #13042

abadger committed:
Fix unittest dailure with the new squash items code

abadger committed:
Add unit tests for issue #12976

bcoca committed:
improved the message when there is no module path override

bcoca committed:
fixed snippet display, short_desc is a string, not a list

bcoca committed:
switched host patterns to use sets, simplified logic which now uses buitins

bcoca committed:
fixed append to add

bcoca committed:
Revert "fixed append to add"
good change on top of bad
This reverts commit 699f344e17e029302af0d5636d6e02e6b8d3dd4c.

bcoca committed:
Revert "switched host patterns to use sets, simplified logic which now uses buitins"
need this to be lists, for order and other considerations
This reverts commit 8e2f0b3f2c9b4ce31ea93701fe9a878d00d92cb3.

bcoca committed:
fixed check mode change response in template
fixes #13049

Etherdaemon committed:
Merge branch 'devel' into fix_boto3_aws_profile_13010

Etherdaemon commented:
No worries - I'll try and fix it up later today.

Etherdaemon committed:
Merge branch 'fix_boto3_aws_profile_13010' of github.com:Etherdaemon/ansible into fix_boto3_aws_profile_13010

Etherdaemon commented:
Hmm - massive fail. Might be easier to close and reopen a new PR instead with just the 1 change.

willthames commented:
@Etherdaemon for future reference

```
git fetch ansibleremote devel
git rebase -i devel
```

change all but the first of the `pick`s to `fixup` (or `f`) (or `squash`/`s`) if you want to incorporate elements of that commit message in your overall commit). 

Then `git push -f yourremote yourbranch`

Etherdaemon commented:
Thanks @willthames. I had screwed it by doing merges and pushes etc. And I was further screwed by my commits being between multiple other peoples commits - so although I think the above was fine, I went ahead and closed this one,created a nice clean one and created a new pull request - https://github.com/ansible/ansible/pull/13053 so that there was no risk of squishing/crapping over other people's commits.
